elements/lisk-transactions/src/utils/verify.ts

```diff
@@ -86,6 +86,24 @@ export const verifyMinRemainingBalance = (
 	return undefined;
 };
 
+export const verifyAccountNonce = (
```

Comment:
we need to detect if it's `===` or `>`, maybe we can have 2 different error for this, so that in transaction pool we can check if it's only one nonce error, it's unprocessable, but if it has more than one error it's invalid?
ie)
```
if (nonce < account.nonce) {
errors.push(new TransactionError());
}
if (nonce !== account.nonce) {
errors.push(new TransactionError());
}
```

Reply:
If we only return one error, in the transaction pool, it would be hard to distinguish (only string comparison) between invalid and unprocessable.
Maybe return 2 errors is better?

Reply:
There can't be two errors, either it's higher or lower nonce. And we don't have different error constructors in transactions to differentiate between errors, so we have to do message comparison.

Reply:
if we return 2 errors for lower nonce and nonce not equal,
we can check
* for non-processable case, `errors.lengh === 1 && errors[0].dataPath === '.nonce'` => not equal error
* for processable `errors.length === 0`
everything else is invalid
dataPath and error length is used to handle exceptions now

Reply:
That's what exactly I mentioned above, adding a check on the length of errors is more fragile and less readable. There can be a bug or logical error in nonce validation which can result in `errors.lenght === 1` which will turn it into invalid execution.  

So be it more clearly readable at the time of processing what are expecting.

Reply:
I added more details to the error object itself. Now we can do something like. 

```
error.dataPath == '.nonce' && error.actual < error.expected
```

Reply:
as long as we dont use message to validate it's good. Another option would be just extend TransactionError and create specific error (I think we had `TransactionPendingError` for multi-sig before